cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/flow/BasicFlowletContext.java

```diff
@@ -83,6 +86,18 @@ public MetricsCollector load(String key) throws Exception {
         }
       });
 
+    this.producerMetrics = CacheBuilder.newBuilder()
```

Comment:
No, because there was an existing cache for the other flowlet metrics. I figured that I should  follow the existing design of that class. But I don't mind at all removing the cache. @chtyim do you know why there was cache in the first place?